src/tests/ftest/control/dmg_telemetry_net.py

```diff
@@ -0,0 +1,154 @@
+                              "returned 0.", retry, cmd)
+
+        # Ignore errors cleaning up mdtest
+        return []
+
+    def tearDown(self):
+        try:
+            super().tearDown()
+        except (CommandFailure, DaosTestError) as excep:
+            self.log.info("%s: tearDown threw an exception "
+                          "because a rank was (intentionally) "
+                          "stopped midway through a run of mdtest.",
+                          repr(excep))
+
+    @skipForTicket("DAOS-9106")
+    def test_net_telemetry(self):
+        """Jira ID: DAOS-9020.
+
+        Test Description: Verify engine net telemetry metrics.
+
+        Use Cases:
+          * Create pool and container.
+          * Use mdtest to create 120K files with 3-way replication.
+          * Stop one server.
+          * Ensure engine_net_ofi_sockets_req_timeout is greater than 0.
+
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=hw,large
+        :avocado: tags=control,telemetry,net
+        :avocado: tags=test_with_telemetry_net,test_net_telemetry
+        """
+
+        # Get the req_timeout metric name, which depends on the libfabric
+        # provider name
+        all_metrics = self.telemetry.get_all_server_metrics_names(
+            self.server_managers[0])
+        p = r"engine_net_\w+_req_timeout"
+        metrics = list(filter(lambda x: search(p, x), all_metrics))
+
+        # Initialize req_timeouts based on what's already populated in the
+        # telemtry system, which may or may not be zero.
+        req_timeouts_start = 0
+
+        data = {}
+        try:
+            data = self.telemetry.get_metrics(metrics[0])
+        except (CommandFailure, DaosTestError) as excep:
+            self.log.info("self.telemetry.get_metrics failed on at least one "
+                          "host with '%s', but may have succeeded elsewhere.",
+                          repr(excep))
+
+        for value in data.values():
+            for metric in value[metrics[0]]["metrics"]:
+                req_timeouts_start += metric["value"]
+
+        req_timeouts_finish = req_timeouts_start
+
+        # set params
+        targets = self.params.get("targets", "/run/server_config/*")
+        rank = self.params.get("rank_to_kill", "/run/testparams/*")
+
+        # create pool
+        self.add_pool(connect=False)
+
+        # make sure pool looks good before we start
+        checks = {
+            "pi_nnodes": len(self.hostlist_servers),
+            "pi_ntargets": len(self.hostlist_servers) * targets,
+            "pi_ndisabled": 0,
+        }
+        self.assertTrue(
+            self.pool.check_pool_info(**checks),
+            "Invalid pool information detected before rebuild")
+
+        # create 1st container
+        self.add_container(self.pool)
+
+        # start mdtest run
+        thread = threading.Thread(target=self.execute_mdtest)
+        thread.start()
+        time.sleep(5)
```

Comment:
We should not be using sleeps in any test method where it can be avoided.  Since we check the pool size information prior to running mdtest, could we use something like a certain percentage of pool size increase as a means of determining when to continue?  Does mdtest need to be running when the rank is stopped?